sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -1201,6 +1201,12 @@ _pi_queue::_pi_queue(std::vector<ze_command_queue_handle_t> &ComputeQueues,
   // fixed to one particular compute CCS (it is so for sub-sub-devices).
   auto &ComputeQueueGroupInfo = Device->QueueGroup[queue_type::Compute];
   ComputeQueueGroup.ZeQueues = ComputeQueues;
+  // Create space to hold immediate commandlists corresponding to the
+  // ZeQueues
+  if (Device->useImmediateCommandLists()) {
+    ComputeQueueGroup.ImmCmdLists = std::vector<pi_command_list_ptr_t>(
+        ComputeQueueGroup.ZeQueues.size(), CommandListMap.end());
```

Comment:
> We still need to initialize the command lists similarly to the regular case. Sub-sub-device properties are handler by indexing into those vectors.

Please show what exactly is broken without this change.

Reply:
This seg. faults:
```
pi_command_list_ptr_t &_pi_queue::pi_queue_group_t::getImmCmdList() {

  uint32_t QueueIndex, QueueOrdinal;
  auto Index = getQueueIndex(&QueueOrdinal, &QueueIndex);

  if (ImmCmdLists[Index] != Queue->CommandListMap.end())          // Here, at ImmCmdLists[Index]
    return ImmCmdLists[Index];
```

on any kernel launch using sub-sub-device or explicit compute_index property on the queue.

Reply:
So this was an existing bug, not incurred by some of the other changes you have in flight?

Reply:
Yes, I first faced it when tried to study our existing sub-sub-device support before writing any new code.